plugin/pkg/shoot/dns/admission.go

```diff
@@ -301,7 +302,11 @@ func assignDefaultDomainIfNeeded(shoot *core.Shoot, projectLister corelisters.Pr
 			if shoot.Spec.DNS == nil {
 				shoot.Spec.DNS = &core.DNS{}
 			}
-			generatedDomain := fmt.Sprintf("%s.%s.%s", shoot.Name, project.Name, domain)
+			shootName := shoot.Name
+			if len(shoot.Name) == 0 && len(shoot.GenerateName) > 0 {
+				shootName = shootregistry.Strategy.NameGenerator.GenerateName(shoot.GenerateName)
```

Comment:
Hm, this name might be different from the name of the actual shoot later on, right? The `GenerateName()` function will be executed again and probably generate a different result?
Should we simply compute a random string here instead of using `.metadata.generateName`?

Reply:
> Hm, this name might be different from the name of the actual shoot later on, right? The `GenerateName()` function will be executed again and probably generate a different result?

Yes. Correct. I digged in check for different hooks on resource admission, but didn't find any solution to get Object with generateName or atleast exact generated name of shoot. 

> Should we simply compute a random string here instead of using `.metadata.generateName`?

Bringing this discussion up was one of the reason for marking this PR WIP. I'm fine with generating random string here itself with generateName prefix. But i thought using name generator would be better idea. Since it will try to follow the same pattern for Shoot name. And it will be single point of code change, if we feel like introducing separate implementation of name generator considering the name validation is dependent on project name.

Do you still want me to change it?

Reply:
Not sure, I'd be confused as end-user if my shoot name ends up with `shoot-foo12` but the DNS is `shoot-bar89`. Then it'd be better to simply call the DNS name `asdjnjknasd28` instead of tampering with the shoot name.
I don't get the point wrt "single point of code change" - what do you mean by that?

Reply:
Makes sense. Then I will generate random string.

>I don't get the point wrt "single point of code change" - what do you mean by that?
Ignore it.